models/migrations/v230.go

```diff
@@ -0,0 +1,204 @@
+	}
+}
+
+func addHeaderAuthorizationEncryptedColWebhook(x *xorm.Engine) error {
+	// Add the column to the table
+	type Webhook struct {
+		ID                 int64 `xorm:"pk autoincr"`
+		RepoID             int64 `xorm:"INDEX"` // An ID of 0 indicates either a default or system webhook
+		OrgID              int64 `xorm:"INDEX"`
+		IsSystemWebhook    bool
+		URL                string `xorm:"url TEXT"`
+		HTTPMethod         string `xorm:"http_method"`
+		ContentType        webhook.HookContentType
+		Secret             string `xorm:"TEXT"`
+		Events             string `xorm:"TEXT"`
+		*webhook.HookEvent `xorm:"-"`
+		IsActive           bool               `xorm:"INDEX"`
+		Type               webhook.HookType   `xorm:"VARCHAR(16) 'type'"`
+		Meta               string             `xorm:"TEXT"` // store hook-specific attributes
+		LastStatus         webhook.HookStatus // Last delivery status
+
+		// HeaderAuthorizationEncrypted should be accessed using HeaderAuthorization() and SetHeaderAuthorization()
+		HeaderAuthorizationEncrypted string `xorm:"TEXT"`
+
+		CreatedUnix timeutil.TimeStamp `xorm:"INDEX created"`
+		UpdatedUnix timeutil.TimeStamp `xorm:"INDEX updated"`
+	}
+	err := x.Sync2(new(Webhook))
+	if err != nil {
+		return err
+	}
+
+	// Migrate the matrix webhooks
+
+	type MatrixMeta struct {
+		HomeserverURL string `json:"homeserver_url"`
+		Room          string `json:"room_id"`
+		MessageType   int    `json:"message_type"`
+	}
+	type MatrixMetaWithAccessToken struct {
+		MatrixMeta
+		AccessToken string `json:"access_token"`
+	}
+
+	err = batchProcess(x,
+		func() *xorm.Session { return x.Where("type=?", "matrix").OrderBy("id") },
+		make([]*Webhook, 0, 50),
+		func(sess *xorm.Session, hook *Webhook) error {
+			// retrieve token from meta
+			var withToken MatrixMetaWithAccessToken
+			err := json.Unmarshal([]byte(hook.Meta), &withToken)
+			if err != nil {
+				return fmt.Errorf("unable to unmarshal matrix meta for webhook[%d]: %w", hook.ID, err)
+			}
+			if withToken.AccessToken == "" {
+				return nil
+			}
+
+			// encrypt token
+			authorization := "Bearer " + withToken.AccessToken
+			hook.HeaderAuthorizationEncrypted, err = secret.EncryptSecret(setting.SecretKey, authorization)
+			if err != nil {
+				return fmt.Errorf("unable to encrypt access token for webhook[%d]: %w", hook.ID, err)
+			}
+
+			// remove token from meta
+			withoutToken, err := json.Marshal(withToken.MatrixMeta)
+			if err != nil {
+				return fmt.Errorf("unable to marshal matrix meta for webhook[%d]: %w", hook.ID, err)
+			}
+			hook.Meta = string(withoutToken)
+
+			// save in database
+			count, err := sess.ID(hook.ID).Cols("meta", "header_authorization_encrypted").Update(hook)
+			if count != 1 || err != nil {
+				return fmt.Errorf("unable to update header_authorization_encrypted for webhook[%d]: %d,%w", hook.ID, count, err)
+			}
+			return nil
+		})
+	if err != nil {
+		return err
+	}
+
+	// Remove access_token from HookTask
+
+	type HookTask struct {
```

Comment:
Just list the changed columns to prevent other PRs have changed some of them but this PR doesn't know.

Reply:
I changed to
```go
	type HookTask struct {
		ID             int64  `xorm:"pk autoincr"`
		PayloadContent string `xorm:"LONGTEXT"`
	}
```

(since no `x.Sync2` is called)